### PR TITLE
Fixes for problems reported by Coverity Scan static analysis

### PR DESCRIPTION
**Pull request by machinekoder**
I ran the Coverity Scan static analyzer over my project [QtQuickVcp](https://github.com/qtquickvcp/qtquickvcp) and detected a few problems in jdns.

Please take a look the implemented fixes.

### CODE REVIEW

**Comment by machinekoder** [comment]
Note: The change of the CMake files is not intentional. I pulled the latest master of this repository before applying the changes.

File: src/qjdns/qjdns.cpp [hunk 1]
```diff
@@ -308,18 +308,22 @@ static void my_srand()
 QJDns::Private::Private(QJDns *_q)
 	: QObject(_q)
 	, q(_q)
+	, mode(QJDns::Multicast)
+	, sess(nullptr)
```
Comment (drizt) [inline src/qjdns/qjdns.cpp]: nullptr - is C++11 feature. JDNS must be compiled with old C++.

File: src/qjdns/qjdns.cpp [hunk 2]
```diff
@@ -138,9 +138,7 @@ static QJDns::Record import_record(const jdns_rr_t *in)
 	out.owner = QByteArray((const char *)in->owner);
 	out.ttl = in->ttl;
 	out.type = in->type;
-	out.rdata = QByteArray((const char *)in->rdata, in->rdlength);
-	out.port = 0;
-	out.weight = 0;
+    out.rdata = QByteArray((const char *)in->rdata, in->rdlength);
```
Comment (drizt) [inline src/qjdns/qjdns.cpp]: Use tabs not spaces.

**Comment by drizt** [comment]
Please avoid CMake changes. It makes me crazy.

**Comment by machinekoder** [comment]
@drizt Should be fixed now

File: src/jdns/jdns.c [hunk 3]
```diff
@@ -2067,6 +2067,10 @@ int _unicast_do_writes(jdns_session_t *s, int now)
 		}
 
 		// in theory, it is not possible for 'ns' to be null here
+		if (ns == 0)
+		{
+			continue; // but lets make sure it will not stab us in the back
```
Comment (Ri0n) [inline src/jdns/jdns.c]: It's possible only if name_servers->count was update to 0 from another thread. Is it really possible ?
If yes please add a comment how and where.

File: src/jdns/jdns_sys.c [hunk 4]
```diff
@@ -271,8 +271,8 @@ static jdns_string_t *file_nextline(FILE *f)
 	at = 0;
 	while(1)
 	{
-		unsigned char c = fgetc(f);
-		if(feof(f))
+		unsigned char c = (unsigned char)fgetc(f); // cast is intentional
+		if(feof(f)) // checking EOF here
```
Comment (Ri0n) [inline src/jdns/jdns_sys.c]: Well, it's clear we could read EOF value from fgetc, but the comments are meaningless. Second one is especially weird.

Comment (machinekoder) [inline src/jdns/jdns_sys.c]: `fgetc` returns a negative value when EOF is reached or an error is encountered. This is the only instance where the cast to unsigned char is not valid. However, thanks to the check for feof afterward this is not an issue here. The comments make sure people see that this is not an error.

File: src/jdns/jdns.c (hunk 3 above)
Comment (machinekoder) [inline src/jdns/jdns.c]: The failure chain is pretty long, I hope the attached trace helps.
image

Comment (Ri0n) [inline src/jdns/jdns.c]: did it ever crash for you here? From what I can see  in the code it is really not possible for 'ns' to be null.

File: src/jdns/jdns_sys.c (hunk 4 above)
Comment (Ri0n) [inline src/jdns/jdns_sys.c]: honestly if one knows how fgetc works then it's clear what will be assigned to c.
and if one does not know fgetc then your comments will say nothing.

Instead I'd add a comment similar to what you just said
// can return EOF(-1),  check it explicitly on the next line

File: src/jdns/jdns.c (hunk 3 above)
Comment (machinekoder) [inline src/jdns/jdns.c]: I had problems with multiple queries on different network infarces (not sure if this is relevant) before I applied the coverity fixes. However, I'm not 100% sure if this was related. Anyway, I think the additional check does not hurt.

**Pull request closed without merging**